SCons/Tool/ninja/__init__.py

```diff
@@ -173,8 +173,7 @@ def generate(env):
         SCons.Warnings.SConsWarning("Failed to import ninja, attempt normal SCons build.")
         return
 
-    env["NINJA_DISABLE_AUTO_RUN"] = GetOption('disable_execute_ninja')
-
+    env["NINJA_DISABLE_AUTO_RUN"] = env.get("NINJA_DISABLE_AUTO_RUN", GetOption('disable_execute_ninja'))
```

Comment:
Was NINJA_DISABLE_AUTO_RUN being set elsewhere pre tool initialization and then ignored? why?

Reply:
Yeah we plan on setting it to True before loading the tool. The mongo build would prefer to set this in the SConscript then from the command line, to maintain the existing workflow.

Reply:
> Yeah we plan on setting it to True before loading the tool. The mongo build would prefer to set this in the SConscript then from the command line, to maintain the existing workflow.

Can't use SetOption() ?

Reply:
it says its not settable

Reply:
> it says its not settable

That's fixable.. I'll take a look tomorrow.
There's a fixed list of what's settable, but you should be able to modify it when the ninja tool is loaded I think. As long as your `SetOption()`  is after you initialize ninja..

Reply:
okay but the GetOption is called during init, so if you SetOption after then it wouldn't reload that in the env var.

I can remove this change if you really want, we will just set the env var after the tool is loaded.